addons/microsoft_calendar_update/__init__.py

```diff
@@ -0,0 +1,21 @@
+from psycopg2 import sql
+
+def update_microsoft_ids(cr, registry):
+    cr.execute("""
+        SELECT 1
+          FROM ir_module_module
+         WHERE name='microsoft_calendar'
+           AND state IN ('installed', 'to upgrade')
+    """)
+    if cr.rowcount:
+        for t in ('calendar_event', 'calendar_recurrence'):
+            cr.execute(
+                sql.SQL(
+                    """
+                    UPDATE {table}
+                    SET microsoft_id = microsoft_id || ':'
+                    WHERE microsoft_id IS NOT NULL
+                    AND microsoft_id NOT LIKE '%:%'
+                    """
+                ).format(table=sql.Identifier(t))
+            )
```

Comment:
What happens if the code is updated without ever installing this?

Reply:
Hum, at the beginning I thought this module was required but now, I think I can remove it by just supporting the old `microsoft_id` format in the `_search_event_id` method.
Will do some tests and remove this module if it works ;-)

Reply:
Ok, I confirm that this module is useless now so I've removed it ;-)